src/azure-cli-core/azure/cli/core/command_recommender.py

```diff
@@ -320,3 +338,73 @@ def _get_error_type(self):
             error_type = AladdinUserFaultType.ValidationError
 
         return error_type.value
+
+    def _extract_parameter_names(self, parameters):  # pylint: disable=no-self-use
+        """Extract parameter names from the raw parameters.
+        An example: ['-g', 'RG', '-n', 'NAME'] ==> ['-g', '-n']
+        """
+
+        from azure.cli.core.commands import AzCliCommandInvoker
+
+        return AzCliCommandInvoker._extract_parameter_names(parameters)  # pylint: disable=protected-access
+
+    def _replace_parameter_values(self, command):
+        """Replace the parameter values in recommended command with values in user's command
+        An example:
+            recommended command: 'az vm create -n MyVm -g MyResourceGroup --image CentOS'
+            user's command:  'az vm create --name user_vm -g user_rg'
+            ==> 'az vm create -n user_vm -g user_rg --image CentOS'
+        """
+
+        def get_parameter_kwargs(parameters):
+            """Get name value mappings from parameter list
+            An example:
+                ['-g', 'RG', '--name=NAME'] ==> {'-g': 'RG', '--name': 'NAME'}
+            """
+
+            parameter_kwargs = dict()
+            for index, parameter in enumerate(parameters):
+                if parameter.startswith('-'):
+
+                    param_name, param_val = parameter, None
+                    if '=' in parameter:
+                        pieces = parameter.split('=')
```

Comment:
there's case param value be empty string "", suggest to be defensive

Reply:
Is it the case like `az vm create -n=`? If so, I think it's aready handled here.